actioncable/app/javascript/action_cable/subscriptions.js

```diff
@@ -0,0 +1,90 @@
+import Subscription from "./subscription"
+
+// Collection class for creating (and internally managing) channel subscriptions. The only method intended to be triggered by the user
+// us ActionCable.Subscriptions#create, and it should be called through the consumer like so:
+//
+//   App = {};
+//   App.cable = ActionCable.createConsumer("ws://example.com/accounts/1");
+//   App.appearance = App.cable.subscriptions.create("AppearanceChannel");
+//
+// For more details on how you'd configure an actual channel subscription, see ActionCable.Subscription.
+export default class Subscriptions {
+  constructor(consumer) {
+    this.consumer = consumer
+    this.subscriptions = []
+  }
+
+  create(channelName, mixin) {
+    const channel = channelName
+    const params = typeof channel === "object" ? channel : {channel}
+    const subscription = new Subscription(this.consumer, params, mixin)
+    return this.add(subscription)
+  }
+
+  // Private
+
+  add(subscription) {
+    this.subscriptions.push(subscription)
+    this.consumer.ensureActiveConnection()
+    this.notify(subscription, "initialized")
+    this.sendCommand(subscription, "subscribe")
+    return subscription
+  }
+
+  remove(subscription) {
+    this.forget(subscription)
+    if (!this.findAll(subscription.identifier).length) {
+      this.sendCommand(subscription, "unsubscribe")
+    }
+    return subscription
+  }
+
+  reject(identifier) {
+    const result = []
+    for (let subscription of this.findAll(identifier)) {
```

Comment:
I'm going to refactor this and the `notify` method below to avoid the use of `for` ... `of` syntax. Babel transpiles `for` ... `of` syntax to use `Symbol.iterator`, which would require a polyfill if your application supports browsers that do not implement `Symbol.iterator`. Running `decaffeinate` produced the `for` ... `of` syntax but in these instances a simpler `forEach` should be sufficient and avoid any `Symbol` issues.